src/util/blockchain.ts

```diff
@@ -126,63 +126,139 @@ export const getBlockchainFromChain = (chainId?: number) => {
-      return Blockchain.Osmosis;
-    case 245: // mainnet
-      return Blockchain.Terra;
-    case 246: // mainnet
-      return Blockchain.CosmosHub;
-    case 247: // mainnet
-      return Blockchain.Juno;
-    case 248: // mainnet
-      return Blockchain.Evmos;
-    case 249: // mainnet
-      return Blockchain.Axelar;
-    case 313: // mainnet
-      return Blockchain.Stride;
-    case 314: // mainnnet
-      return Blockchain.Kujira;
-    case 315: // mainnnet
-      return Blockchain.Terra2;
-    case 316: // mainnnet
-      return Blockchain.Quicksilver;
-    case 317: // mainnnet
-      return Blockchain.Comdex;
-    case 318: // mainnnet
-      return Blockchain.StafiHub;
-    case 319: // mainnnet
-      return Blockchain.Persistence;
-    case 320: // mainnnet
-      return Blockchain.Stargaze;
-    case 42161: // mainnnet
-      return Blockchain.Arbitrum;
-    default:
-      return undefined;
+export const blockchainForChainId = (chainId?: number, network?: Network): Blockchain | undefined => {
+  if (!network) { //default list if no network is specified
+    switch (chainId) {
+      case 0:
+        return Blockchain.Native
+      case 1:
+        return Blockchain.Btc
+      case 2:
+      case 350: // devnet
+        return Blockchain.Ethereum
+      case 4:
+      case 5: // devnet
+        return Blockchain.Neo
+      case 6:
+      case 79: // devnet
+        return Blockchain.BinanceSmartChain
+      case 9: // mainnet
+      case 10:
+      case 18:
+      case 110: // testnet
+      case 111:
+        return Blockchain.Zilliqa
+      case 14: // mainnet
+      case 88: // testnet
+        return Blockchain.Neo3
+      case 19: // mainnet
+        return Blockchain.Arbitrum
+      case 244: // mainnet
+        return Blockchain.Osmosis
+      case 245: // mainnet
+        return Blockchain.Terra
+      case 246: // mainnet
+        return Blockchain.CosmosHub
+      case 247: // mainnet
+        return Blockchain.Juno
+      case 248: // mainnet
+        return Blockchain.Evmos
+      case 249: // mainnet
+        return Blockchain.Axelar
+      case 313: // mainnet
+        return Blockchain.Stride
+      case 314: // mainnnet
+        return Blockchain.Kujira
+      case 315: // mainnnet
+        return Blockchain.Terra2
+      case 316: // mainnnet
+        return Blockchain.Quicksilver
+      case 317: // mainnnet
+        return Blockchain.Comdex
+      case 318: // mainnnet
+        return Blockchain.StafiHub
+      case 319: // mainnnet
+        return Blockchain.Persistence
+      case 320: // mainnnet
+        return Blockchain.Stargaze
+      case 42161: // mainnnet
+        return Blockchain.Arbitrum
+      default:
+        return undefined
+    }
+  } else {
+    switch (network) {
+      case Network.MainNet:
+        switch (chainId) {
+          case 0:
+            return Blockchain.Native
+          case 1:
+            return Blockchain.Btc
+          case 4:
+            return Blockchain.Carbon
```

Comment:
On Carbon chain, chain-id 4 is Neo Legacy (refer to [Bridges API](https://api.carbon.network/carbon/coin/v1/bridges?pagination.limit=10000)). Where did you get chain-id 4 for Carbon?
<img width="494" alt="Screenshot 2023-01-31 at 9 10 53 PM" src="https://user-images.githubusercontent.com/65583522/215768971-5618e498-9b21-4149-ab01-11161a2bf0c8.png">